### ISSUE TITLE
create kernel ipsec-interface (pseudo device) during orient? (but not add addresses)

### ISSUE THREAD

**Issue by cagney** [libreswan/libreswan#1807]
This follows up

- [ ] #1795

@bleve suggested, off line, creating the kernel ipsec-interface when the connection is being added except that doesn't work because things aren't yet oriented.

So the next alternative, clearly :-), is to create the kernel ipsec-interface during orient (and presumably toss it when disorienting).
This would also suggest a split between creating an kernel ipsec-interface and populating it with addresses (the latter possibly delayed until in `route.[hc]`, but it depends on `interface-ip=` and/or` sourceip=`).

@wofferl @antonyantony thoughts

**Comment by cagney** [comment]
somewhat supporting this idea, the test `ipsec-interface-15-blocked-ikev2-xfrmi` fails with:
```
"westnet6-eastnet6": added IKEv2 connection
"westnet4-eastnet4" #5: initiating IKEv2 connection to 192.1.2.23 using UDP
"westnet4-eastnet4" #5: sent IKE_SA_INIT request to 192.1.2.23:UDP/500
"westnet4-eastnet4" #5: processed IKE_SA_INIT response from 192.1.2.23:UDP/500 {cipher=AES_GCM_16_256 integ=n/a prf=HMAC_SHA2_512 group=DH19}, initiating IKE_AUTH
"westnet4-eastnet4" #5: sent IKE_AUTH request to 192.1.2.23:UDP/500
"westnet4-eastnet4" #5: initiator established IKE SA; authenticated peer using authby=secret and ID_FQDN '@east'
ERROR: "westnet4-eastnet4" #6: ipsec-interface ipsec1 with ID 1 exists but is not a valid: IFLA_XFRM_LINK attribute is zero, check 'ip -d link show dev ipsec1'
"westnet4-eastnet4" #6: CHILD SA failed: TEMPORARY_FAILURE
"westnet4-eastnet4" #5: IKE SA established but initiator rejected Child SA response
"westnet4-eastnet4" #6: sent INFORMATIONAL request to delete larval Child SA using IKE SA #5
ERROR: "westnet4-eastnet4" #6: ipsec-interface ipsec1 with ID 1 exists but is not a valid: IFLA_XFRM_LINK attribute is zero, check 'ip -d link show dev ipsec1'
EXPECTATION FAILED: "westnet4-eastnet4" #6: (*child) == ((void *)0) (teardown_child() +1238 programs/pluto/routing.c)
```
because the code doesn't know how to recover - it's dug too deep a hole.  Creating the kernel ipsec-interface would instead block orientation.  This code would add the address which shouldn't fail.

**Comment by cagney** [comment]
for this to work, reorienting needs to deference the old kernel interface before adding the new one
(also note that the routing code's call to `remove_kernel_ipsec_interface()` doesn't delete the interface, just the address)

**Comment by cagney** [comment]
Another reason to not add kernel ipsec-interfaces before orienting is that the "link" or physical device isn't known; and re-orienting can change it.

**Comment by cagney** [comment]
merged 72cf42574b0f7f48220442ab2a7e576a73ef2d32
